binance-pipeline/lambdas/tests/binance/test_binance_utils.py

```diff
@@ -0,0 +1,62 @@
+# ./tests/binance/test_binance_utils.py
+from datetime import datetime
+import unittest
+#
+from binance.binance_utils import get_date
+from binance.binance_utils import TradeIDFinder
+
+class TestTradeIDFinder(unittest.TestCase):
+    symbol = 'BNBBTC'
+    trade_id_finder = TradeIDFinder(symbol, datetime.now())
```

Comment:
Like here. Recommend that you pass the API object as a dependency to TradeIDFinder and inject a mock for unit tests.

Reply:
Okay that's reasonably simple to refactor I will do that

Could TradeIDFinder ever be seen as as extension of the base API object and should actually extend/inherit from it instead?

Reply:
Sounds like that would introduce coupling. I'm a fan of the dependency injection approach, so you can more easily compose functionality, but see what you feel.